File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter04/ClientActivated/MathLibrary/Customer.vb
Public Class Customer
   Inherits MarshalByRefObject

   Private mName As String

   Public Sub New()
      Console.WriteLine("Customer.ctor()")
   End Sub

   Public Sub New(name As String)
      Console.WriteLine("Customer.ctor({0})", name)
      mName = name
   End Sub
   
   Public Function SayHello() As String
      Console.WriteLine("Customer.SayHello()")
      Return "Hello " & mName
   End Function
    
End Class

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter04/Windows_Service/MathLibrary/Customer.cs
using System;

namespace MathLibrary
{
   public class Customer : MarshalByRefObject
   {
      string mName;

      public Customer()
      {
         Console.WriteLine("Customer.ctor()");
      }

      public Customer(string name)
      {
         Console.WriteLine("Customer.ctor({0})", name);
         mName = name;
      }

      public string SayHello()
      {
         Console.WriteLine("Customer.SayHello()");
         return "Hello " + mName;
      }
   }
}
